cli/command/task.go

```diff
@@ -10,6 +17,220 @@ func NewTaskCommand(b *CommandBase) *TaskCommand {
 }
 
-func (e *TaskCommand) Command() cli.Command {
-	return cli.Command{}
+func (t *TaskCommand) Command() cli.Command {
+	return cli.Command{
+		Name:  "task",
+		Usage: "manage layer0 tasks",
+		Subcommands: []cli.Command{
+			{
+				Name:      "create",
+				Usage:     "create a new task",
+				Action:    t.create,
+				ArgsUsage: "ENVIRONMENT TASK_NAME DEPLOY",
+				Flags: []cli.Flag{
+					cli.IntFlag{
+						Name:  "copies",
+						Value: 1,
+						Usage: "number of copies of deploy to run (default: 1)",
+					},
+					cli.StringSliceFlag{
+						Name:  "env",
+						Usage: "environment variable override in format 'CONTAINER:VAR=VAL' (can be specified multiple times)",
+					},
+				},
+			},
+			{
+				Name:      "delete",
+				Usage:     "delete a Task",
+				ArgsUsage: "TASK_NAME",
+				Action:    t.delete,
+			},
+			{
+				Name:      "list",
+				Usage:     "list all Tasks",
+				Action:    t.list,
+				ArgsUsage: " ",
+			},
+			{
+				Name:      "read",
+				Usage:     "describe a Task",
+				Action:    t.read,
+				ArgsUsage: "TASK_NAME",
+			},
+			{
+				Name:      "logs",
+				Usage:     "get the logs for a task",
+				Action:    t.logs,
+				ArgsUsage: "TASK_NAME",
+				Flags: []cli.Flag{
+					cli.IntFlag{
+						Name:  "tail",
+						Usage: "number of lines from the end to return",
+					},
+					cli.StringFlag{
+						Name:  "start",
+						Usage: "the start of the time range to fetch logs (format: YYYY-MM-DD HH:MM)",
+					},
+					cli.StringFlag{
+						Name:  "end",
+						Usage: "the end of the time range to fetch logs (format: YYYY-MM-DD HH:MM)",
+					},
+				},
+			},
+		},
+	}
+}
+
+func (t *TaskCommand) create(c *cli.Context) error {
+	args, err := extractArgs(c.Args(), "ENVIRONMENT", "TASK_NAME", "DEPLOY")
+	if err != nil {
+		return err
+	}
+
+	overrides, err := parseOverrides(c.StringSlice("env"))
+	if err != nil {
+		return err
+	}
+
+	environmentID, err := t.resolveSingleEntityIDHelper("environment", args["ENVIRONMENT"])
+	if err != nil {
+		return err
+	}
+
+	deployID, err := t.resolveSingleEntityIDHelper("deploy", args["DEPLOY"])
+	if err != nil {
+		return err
+	}
+
+	req := models.CreateTaskRequest{
+		TaskName:           args["TASK_NAME"],
+		EnvironmentID:      environmentID,
+		DeployID:           deployID,
+		ContainerOverrides: overrides,
+	}
+
+	jobID, err := t.client.CreateTask(req)
+	if err != nil {
+		return err
+	}
+
+	return t.waitOnJobHelper(c, jobID, "creating", func(taskID string) error {
+		task, err := t.client.ReadTask(taskID)
+		if err != nil {
+			return err
+		}
+
+		return t.printer.PrintTasks(task)
+	})
+}
+
+func (t *TaskCommand) delete(c *cli.Context) error {
+	return t.deleteHelper(c, "task", func(taskID string) (string, error) {
+		return t.client.DeleteTask(taskID)
+	})
+}
+
+func (t *TaskCommand) list(c *cli.Context) error {
+	taskSummaries, err := t.client.ListTasks()
+	if err != nil {
+		return err
+	}
+
+	return t.printer.PrintTaskSummaries(taskSummaries...)
```

Comment:
[nit] we may want to spin this off into a separate issue, but i've found during testing (and experience) that this command can take quite a while to return, especially when there are many tasks. I think we may want to consider adding the spinner behavior for this as a result, and possibly other `list` operations as well.

Reply:
Yeah I think it should be a separate issue, but if there is a long wait we should change this.

Reply:
I'm hoping with the updated api changes this command will actually be pretty fast. I haven't tested with a ton of tasks yet, but hoping this will be a non-issue